lucene/core/src/java/org/apache/lucene/index/OrdinalMap.java

```diff
@@ -271,13 +273,26 @@ protected boolean lessThan(TermsEnumIndex a, TermsEnumIndex b) {
       globalOrd++;
     }
 
-    this.firstSegments = firstSegments.build();
-    this.globalOrdDeltas = globalOrdDeltas.build();
+    long ramBytesUsed = BASE_RAM_BYTES_USED + segmentMap.ramBytesUsed();
+    this.valueCount = globalOrd;
+
+    // If the first segment contains all of the global ords, then we can apply a small optimization
+    // and hardcode the first segments and global ord deltas as all zeroes.
+    if (ordDeltaBits.length > 0 && ordDeltaBits[0] == 0L && ordDeltas[0].size() == this.valueCount) {
+      this.firstSegments = LongValues.ZEROES;
+      this.globalOrdDeltas = LongValues.ZEROES;
+      ramBytesUsed += RamUsageEstimator.shallowSizeOf(LongValues.ZEROES);
```

Comment:
We could ignore it completely from `ramBytesUsed`, since this singleton is allocated anyway, regardless of whether the optimization uses it.

Reply:
I added this to address a failure in `TestOrdinalMap`. But now I see it makes more sense to modify the test !